x/programs/rust/wasmlanche-sdk/src/program.rs

```diff
@@ -6,9 +6,30 @@ use crate::{
     Gas,
 };
 use borsh::{BorshDeserialize, BorshSerialize};
-use std::{cell::RefCell, collections::HashMap};
+use std::{cell::RefCell, collections::HashMap, io::Read};
 use thiserror::Error;
 
+#[cfg_attr(feature = "debug", derive(Debug))]
+pub struct DeferDeserialize(Vec<u8>);
```

Comment:
Can you add a warning doc-comment stating that "since the `Reader` in the `BorshDeserialize::deserialize_reader` function doesn't provide any size-hint, it's possible that this type performs multiple allocations during deserialization"? And also add that "it should be used sparingly."

Reply:
Added a doc comment on the `deserialize_reader` function directly. I couldn't get a working hyperlink to the function in a warning div.

Reply:
Can you move those docs here? The type itself should be used sparingly. But the docs you wrote below are perfect (especially that warning)

Reply:
I have split the warning text and have framed those two with the warning.